lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -0,0 +1,167 @@
+    """
+    SubscriberDBCloudClient for requesting subscribers from Orchestrator.
+
+    SubscriberDBCloudClient calls the Orchestrator's SubscriberDBCloud service
+    to fetch pages of subscribers stored in Orchestrator and updates the local
+    subscriberdb store with the changes.
+    """
+    DEFAULT_SUBSCRIBERDB_SYNC_INTERVAL = 180
+    DEFAULT_SUBSCRIBER_PAGE_SIZE = 250
+    SUBSCRIBERDB_REQUEST_TIMEOUT = 10
+
+    def __init__(self,
+                 loop: asyncio.AbstractEventLoop,
+                 store: SqliteStore,
+                 subscriber_page_size: int,
+                 sync_interval: int,
+                 grpc_client_manager: GRPCClientManager):
+
+        super().__init__(
+            sync_interval,
+            loop
+        )
+        self._loop = loop
+        self._subscriber_page_size = subscriber_page_size
+        self._store = store
+
+        # grpc_client_manager to manage grpc client recycling
+        self._grpc_client_manager = grpc_client_manager
+
+    async def _run(self) -> None:
+        subscribers = await self._get_subscribers()
+        if subscribers is not None:
+            self._process_subscribers(subscribers)
+
+    async def _get_subscribers(self) -> [SubscriberData]:
+        subscriberdb_cloud_client = self._grpc_client_manager.get_client()
+        subscribers = []
+        page_token = ""
+        sync_start = datetime.datetime.now()
+        while True:
+            try:
+                req = ListSubscribersRequest(
+                    page_size=self._subscriber_page_size,
+                    page_token=page_token)
+                response = await grpc_async_wrapper(
+                    subscriberdb_cloud_client.ListSubscribers.future(
+                        req,
+                        self.SUBSCRIBERDB_REQUEST_TIMEOUT
+                    ),
+                    self._loop)
+                subscribers.extend(response.subscribers)
+                if response.next_page_token == "":
+                    logging.info("Successfully fetched all subscriber "
+                                 "pages from the cloud!")
+                    SUBSCRIBER_SYNC_SUCCESS_TOTAL.inc()
+                    time_elapsed = datetime.datetime.now() - sync_start
+                    SUBSCRIBER_SYNC_LATENCY.observe(
+                        time_elapsed.total_seconds() * 1000)
+                    break
+                page_token = response.next_page_token
+            except grpc.RpcError as err:
+                logging.error("Fetch subscribers error! [%s] %s", err.code(),
+                              err.details())
+                time_elapsed = datetime.datetime.now() - sync_start
+                SUBSCRIBER_SYNC_LATENCY.observe(
+                    time_elapsed.total_seconds() * 1000)
+                SUBSCRIBER_SYNC_FAILURE_TOTAL.inc()
+                return None
+
+        return subscribers
+
+    def _process_subscribers(self, subscribers: SubscriberData) -> None:
```

Comment:
Maybe `_install_subscribers`, either is fine

Reply:
Leaving as `process` as we delete some and install others